src/Http/ApiProblem/ApiProblemFactory.php

```diff
@@ -77,23 +79,22 @@ public static function createFromThrowable(
             case $e instanceof DocumentDoesNotExist:
             case $e instanceof EntityNotFoundException:
             case $e instanceof MediaObjectNotFoundException:
-                $routeParameters = new RouteParameters($request);
-                if ($routeParameters->hasEventId()) {
+                if ($routeParameters && $routeParameters->hasEventId()) {
```

Comment:
I think by extracting this case statement into a separate method, like you did with the cultureFeedException, you can just check once if you have route parameters and make the code even more readable.

Reply:
Or use an early return with `return ApiProblem::urlNotFound();` when `$routeParameters` are `null`?

Reply:
I added the early return in [a85df3c](https://github.com/cultuurnet/udb3-silex/pull/1186/commits/a85df3ce937620bf63d829f16536cfcff0728818). If I'd extract it to a separate method I'd probably need an early return as well to avoid the duplicate checks. So I didn't extract it for now but we can definitely do that when this logic becomes too big.